dias/analyzers/sensitivity_analyzer.py

```diff
@@ -0,0 +1,273 @@
+from dias import CHIMEAnalyzer
+from datetime import datetime
+import calendar
+from caput import config
+from dias.utils.string_converter import str2timedelta, datetime2str
+from ch_util import data_index
+
+import os
+import sys
+import glob
+import time
+import subprocess
+import gc
+
+import h5py
+import numpy as np
+import scipy.constants
+
+from collections import Counter
+from collections import defaultdict
+
+from ch_util import andata
+from ch_util import tools
+from ch_util import ephemeris
+
+
+class SensitivityAnalyzer(CHIMEAnalyzer):
+    """SensitivityAnalyzer.
+    This subclass of dias.analyzer.Analyzer describes the analyzer.
+    """
+    acq_dir         = config.Property(proptype=str,   default='/mnt/gong/archive')
+    period          = config.Property(proptype=str,  default="24h")
+    start_time      = config.Property(proptype=str,   default='2018-01-03 17:13:13')
+
+    correlator      = config.Property(proptype=str,   default='chime')
+    output_suffix   = config.Property(proptype=str,   default='sensitivity')
+    acq_suffix      = config.Property(proptype=str,   default='stack_corr')
+
+    nfreq_per_block = config.Property(proptype=int,   default=16)
+    include_auto    = config.Property(proptype=bool,  default=False)
+    include_intracyl= config.Property(proptype=bool,  default=False)
+    include_crosspol= config.Property(proptype=bool,  default=False)
+    sep_cyl         = config.Property(proptype=bool,  default=False)
+    cyl_start_char  = config.Property(proptype=int,  default=65)
+    cyl_start_num   = config.Property(proptype=int,  default=2)
+
+    def setup(self):
+        """Setup stage: this is called when dias starts up."""
+        self.logger.info('Started computation of telescope sensitvity...')
+
+    def run(self):
+        """Main task stage: analyze data from the last period.
+        """
+        # Setup logging
+
+        stop_time  = datetime.utcnow()  
+        start_time = stop_time - str2timedelta(self.period) #Query files from now to period hours back 
```

Comment:
Remember that data produced in the last 2-3 hours is probably not in the archive yet, when this runs. The data that is archived in these hours will thus never be seen by your analyzer.

Reply:
Thanks. I have added a new property `lag` to query data with no gaps.